crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -160,25 +60,41 @@ impl AuthorityStorePruner {
         Ok(pruned)
     }
 
-    fn process_checkpoints(
+    fn prune_objects(
         perpetual_db: &Arc<AuthorityPerpetualTables>,
         checkpoint_store: &Arc<CheckpointStore>,
         num_epochs_to_retain: u64,
     ) -> anyhow::Result<()> {
-        let mut pruned_seq_number = checkpoint_store
+        let mut checkpoint_number = checkpoint_store
             .get_highest_pruned_checkpoint_seq_number()?
             .unwrap_or_default();
+        let mut checkpoint_digest = CheckpointDigest::random();
         let current_epoch = checkpoint_store
             .get_highest_executed_checkpoint()?
             .map(|c| c.epoch())
             .unwrap_or_default();
+        let mut checkpoints_in_batch = 0;
+        let mut batch_effects = vec![];
+        let mut network_total_transactions = 0;
+
+        debug!(
+            "Starting object pruning. Current epoch: {}. Latest pruned checkpoint: {}",
+            current_epoch, checkpoint_number
+        );
         loop {
```

Comment:
I think this loop tries to get all checkpoints for past epoch. I am sure (at least we should) at this point have a table of epoch -> checkpoint number we can use to identify this range, and then use an iterator and multi-get to get all checkpoints at once? Doing one by one in a blocking fashion is not very databasey for such a bulk operation.

Reply:
good point. It's done mostly to support case of num_epochs_to_retain = 0. We're doing batching on application layer, which is suboptimal. We can apply this optimization later if we get rid of option of aggressive pruner